src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -38,13 +38,29 @@ internal sealed class InMethodBinder : LocalScopeBinder
 #endif
 
         public InMethodBinder(MethodSymbol owner, Binder enclosing)
-            : base(enclosing, enclosing.Flags & ~BinderFlags.AllClearedAtExecutableCodeBoundary)
+            : base(enclosing, ConstructFlags(owner, enclosing))
         {
             Debug.Assert(!enclosing.Flags.Includes(BinderFlags.InCatchFilter));
             Debug.Assert((object)owner != null);
             _methodSymbol = owner;
         }
 
+        private static BinderFlags ConstructFlags(MethodSymbol owner, Binder enclosing)
+        {
+            BinderFlags flags = enclosing.Flags & ~BinderFlags.AllClearedAtExecutableCodeBoundary;
+
+            if (owner.IsIterator && enclosing.Compilation.IsFeatureEnabled(MessageID.IDS_FeatureRefUnsafeInIteratorAsync))
+            {
+                // Spec §13.3.1: "An iterator block always defines a safe context,
```

Comment:
>§13.3.1 [](http://example.com/codeflow?start=24&length=7)

Paragraph numbers are not stable, consider using/including the title of the relevant section instead..